choreographer/src/main/java/eu/arrowhead/core/choreographer/database/service/ChoreographerSessionDBService.java

```diff
@@ -1,237 +1,302 @@
-//      logger.debug("finalizeSession started...");
-//
-//      try {
-//          Optional<ChoreographerSession> sessionOptional = choreographerSessionRepository.findById(sessionId);
-//          if (sessionOptional.isPresent()) {
-//              ChoreographerSession session = sessionOptional.get();
-//              session.setStatus(ChoreographerStatusType.DONE);
-//              createWorklog("Session with ID of " + sessionId + " finished successfully.", "");
-//              return choreographerSessionRepository.saveAndFlush(session);
-//          } else {
-//              throw new InvalidParameterException("Session with given ID doesn't exist.");
-//          }
-//      } catch (InvalidParameterException ex) {
-//          throw ex;
-//      } catch (final Exception ex) {
-//          logger.debug(ex.getMessage(), ex);
-//          throw new ArrowheadException(CoreCommonConstants.DATABASE_OPERATION_EXCEPTION_MSG);
-//      }
-//  }
+    //-------------------------------------------------------------------------------------------------
+    @Transactional(rollbackFor = ArrowheadException.class)
+    public ChoreographerSession changeSessionStatus(final long sessionId, final ChoreographerSessionStatus status) {
+        logger.debug("changeSessionStatus started...");
+        Assert.notNull(status, "ChoreographerSessionStatus is null");
 
-//  //-------------------------------------------------------------------------------------------------
-//  @Transactional(rollbackFor = ArrowheadException.class)
-//  public ChoreographerSession setSessionStatus(final long sessionId, final ChoreographerStatusType state) {
-//      logger.debug("changeSessionState started...");
-//
-//      if (state == null) {
-//          throw new InvalidParameterException("State is null or blank.");
-//      }
-//
-//      try {
-//          Optional<ChoreographerSession> sessionOptional = choreographerSessionRepository.findById(sessionId);
-//          if (sessionOptional.isPresent()) {
-//              ChoreographerSession session = sessionOptional.get();
-//              session.setStatus(state);
-//              createWorklog("New status of session with ID of " + sessionId + ": " + state, "");
-//              return choreographerSessionRepository.saveAndFlush(session);
-//          } else {
-//              throw new InvalidParameterException("Session with given ID doesn't exist.");
-//          }
-//      } catch (InvalidParameterException ex) {
-//          throw ex;
-//      } catch (final Exception ex) {
-//          logger.debug(ex.getMessage(), ex);
-//          throw new ArrowheadException(CoreCommonConstants.DATABASE_OPERATION_EXCEPTION_MSG);
-//      }
-//  }
+        try {
+      	  final Optional<ChoreographerSession> optional = sessionRepository.findById(sessionId);
+      	  if (optional.isEmpty()) {
+      		  worklogAndThrow("Session status change has been failed", new InvalidParameterException("Session with id " + sessionId + " not exists"));
```

Comment:
worklog message should contains the cause too

Reply:
ok, now I see that the exception message are in the DB too, so the cause is there.

So, please ignore all the 'casue in worklog' comments